### PR TITLE
Profile: Add Gas Description to Disambiguate.

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Changes made:

1. add the gas description to the label on pressure graphs to disambiguate if multiple identical gasmixes are shown;
2. move the label to the right, where the end pressures will typically be more spread out than the starting pressures.

### Additional information:

image

### CODE REVIEW

File: profile-widget/diveprofileitem.cpp
Comment (bstoeger) [inline profile-widget/diveprofileitem.cpp:644]: I know that you only moved code here and not that it matters at all, but the 'pure' thing to do is `QStringLiteral()`, as the format string will then not be constructed at runtime.

That said, the more you work on the profile, the more it becomes a pressing issue to get a resolution for #3863. Rebasing is becoming very painful and error-prone. :(

File: profile-widget/diveprofileitem.cpp
Comment (bstoeger) [inline profile-widget/diveprofileitem.cpp:658]: Not sure, but I think if both arguments are `QString`s, `.arg(s1, s2)` is preferred. `.arg(s1).arg(s2)` does some funky rewriting of the format string (replace %2 by %1, etc).

**Review by bstoeger — APPROVED** [review]

File: profile-widget/diveprofileitem.cpp
Comment (bstoeger) [reply profile-widget/diveprofileitem.cpp:644]: PS: of course the biggest pain is self-inflicted with my porting to C++.

Comment (mikeller) [reply profile-widget/diveprofileitem.cpp:644]: @bstoeger: Yes, I suspect that #3863 will be a rebase from hell, mostly due to changes that I made. But a lot of these changes were motivated by bug reports or shortcomings reported by users, and I don't think holding off this kind of thing for months on end is helpful if we want to keep our users engaged with helping us by providing feedback.
Is there anything that can be done to unblock you on #3863?

Comment (bstoeger) [reply profile-widget/diveprofileitem.cpp:644]: > Yes, I suspect that https://github.com/subsurface/subsurface/pull/3863 will be a rebase from hell, mostly due to changes that I made.

No worries, the worst problem will be my own conversions to C++.

I'm stuck with the link-order issues on mobile. The only fix I can think about is moving all the chart/QtQuick related stuff in a single "librabry". :( Then it needs of course testing on mobile, with pinch-zooming, etc.

Currently, conversion to C++ has priority for me, though.

**Pull request merged**
